packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/OutputTypes/MrwSerializationTypeProvider.cs

```diff
@@ -0,0 +1,387 @@
+            (
+              new MethodSignature(nameof(IPersistableModel<object>.Create), null, null, MethodSignatureModifiers.None, typeOfT, null, new[] { dataParameter, _serializationOptionsParameter }, ExplicitInterface: _iPersistableModelTInterface),
+              // TO-DO: Call the base model ctor for now until the model properties are serialized https://github.com/microsoft/typespec/issues/3330
+              Return(new NewInstanceExpression(typeOfT, Array.Empty<ValueExpression>()))
+            );
+        }
+
+        /// <summary>
+        /// Builds the I model GetFormatFromOptions method.
+        /// </summary>
+        internal CSharpMethod BuildIModelGetFormatFromOptionsMethod()
+        {
+            ValueExpression jsonWireFormat = SystemSnippets.JsonFormatSerialization;
+            // ModelReaderWriterFormat IPersistableModel<T>.GetFormatFromOptions(ModelReaderWriterOptions options)
+            return new CSharpMethod
+            (
+              new MethodSignature(nameof(IPersistableModel<object>.GetFormatFromOptions), null, null, MethodSignatureModifiers.None, typeof(string), null, new[] { _serializationOptionsParameter }, ExplicitInterface: _iPersistableModelTInterface),
+              jsonWireFormat
+            );
+        }
+
+        /// <summary>
+        /// Builds the serialization constructor for the model.
+        /// </summary>
+        /// <param name="baseinitializationCtor">The base initialization constructor for the model.</param>
+        /// <returns>The constructed serialization constructor. Otherwise <c>null</c> is returned.</returns>
+        internal CSharpMethod? BuildSerializationConstructor(CSharpMethod? baseinitializationCtor)
+        {
+            var baseConstructorParameters = new List<Parameter>();
+            if (baseinitializationCtor != null)
+            {
+                baseConstructorParameters.AddRange(baseinitializationCtor.Signature.Parameters);
+            }
+
+            var serializationCtorParameters = BuildSerializationConstructorParameters();
+            bool serializationParametersMatchInitialization = baseConstructorParameters.SequenceEqual(
+                serializationCtorParameters, Parameter.EqualityComparerByType);
+
+            if (serializationParametersMatchInitialization)
+            {
+                return null;
+            }
+
+            return new CSharpMethod(
+                signature: new ConstructorSignature(
+                    Type,
+                    $"Initializes a new instance of {Type:C}",
+                    null,
+                    MethodSignatureModifiers.Internal,
+                    serializationCtorParameters),
+                bodyStatements: new MethodBodyStatement[]
+                {
+                    GetPropertyInitializers(serializationCtorParameters)
+                });
+        }
+
+        private MethodBodyStatement GetPropertyInitializers(IReadOnlyList<Parameter> parameters)
```

Comment:
This is a duplicate that ModelTypeProvider uses as well and should probably be a method we can share.

It seems like the conversion is from a PropertyDeclaration to a MethodBodyStatment.

I wonder if we should have a method on PropertyDeclaration called ToInitializationStatement() which does the conversion and then here we just loop through the properties and call it?